mmaction/integration/nncf/compression.py

```diff
@@ -0,0 +1,288 @@
+    try:
+        from nncf.torch.initialization import DefaultInitializingDataLoader
+        from nncf.torch.nncf_network import NNCFNetwork
+        
+        class_InitializingDataLoader = DefaultInitializingDataLoader
+    except ImportError:
+        raise RuntimeError(
+            'Cannot import the standard functions of NNCF library '
+            '-- most probably, incompatible version of NNCF. '
+            'Please, use NNCF version pointed in the documentation.')
+else:
+    class DummyInitializingDataLoader:
+        pass
+
+
+    class_InitializingDataLoader = DummyInitializingDataLoader
+
+
+class MMInitializeDataLoader(class_InitializingDataLoader):
+    def get_inputs(self, dataloader_output):
+        return (), dataloader_output
+
+
+def get_nncf_metadata():
+    """
+    The function returns NNCF metadata that should be stored into a checkpoint.
+    The metadata is used to check in wrap_nncf_model if the checkpoint should be used
+    to resume NNCF training or initialize NNCF fields of NNCF-wrapped model.
+    """
+    check_nncf_is_enabled()
+    return dict(nncf_enable_compression=True, nncf_version=get_nncf_version())
+
+
+def is_checkpoint_nncf(path):
+    """
+    The function uses metadata stored in a checkpoint to check if the
+    checkpoint was the result of trainning of NNCF-compressed model.
+    See the function get_nncf_metadata above.
+    """
+    try:
+        checkpoint = torch.load(path, map_location='cpu')
+        meta = checkpoint.get('meta', {})
+        nncf_enable_compression = meta.get('nncf_enable_compression', False)
+        return bool(nncf_enable_compression)
+    except FileNotFoundError:
+        return False
+
+def get_nncf_config_from_meta(path):
+    """
+    The function uses metadata stored in a checkpoint to restore the nncf
+    part of the model config.
+    """
+    logger = get_root_logger()
+    checkpoint = torch.load(path, map_location='cpu')
+    meta = checkpoint.get('meta', {})
+
+    nncf_enable_compression = meta.get('nncf_enable_compression', False)
+    assert nncf_enable_compression, \
+            'get_nncf_config_from_meta should be run for NNCF-compressed checkpoints only'
+
+    config_text = meta['config']
+
+    with tempfile.NamedTemporaryFile(prefix='config_', suffix='.py',
+                                     mode='w', delete=False) as f_tmp:
+        f_tmp.write(config_text)
+        tmp_name = f_tmp.name
+    cfg = mmcv.Config.fromfile(tmp_name)
+    os.unlink(tmp_name)
+
+    nncf_config = cfg.get('nncf_config')
+
+    assert isinstance(nncf_config, dict), (
+            f'Wrong nncf_config part of the config saved in the metainfo'
+            f' of the snapshot {path}:'
+            f' nncf_config={nncf_config}')
+
+    nncf_config_part = {
+            'nncf_config': nncf_config,
+            'find_unused_parameters': True
+            }
+    if nncf_config_part['nncf_config'].get('log_dir'):
+        # TODO(LeonidBeynenson): improve work with log dir
```

Comment:
@LeonidBeynenson, Do you really have a plan to improve it (it means we will wait for the PR)? If not we can remove it.

Reply:
No, I do not think that we improve it in the nearest future.
So, I think we could remove it
```suggestion
```